test-interface/src/main/scala/org/scalajs/testinterface/TestUtils.scala

```diff
@@ -1,32 +1,31 @@
-package org.scalajs.testinterface
+package org.scalajs
+package testinterface
 
-import scala.scalanative.testinterface.PreloadedClassLoader
+import scala.scalanative.reflect.Reflect
 
-// ScalaJS defines this to mimic reflection.
-// We substitute it with our implementation, expecting that
-// the `ClassLoader` that we'll receive is a `PreloadedClassLoader`,
-// from which we'll directly get the instance we're interested in.
+@deprecated(message = "Use scala.scalanative.reflect.Reflect instead.",
+            since = "0.4.0")
 object TestUtils {
 
-  def newInstance(name: String, loader: ClassLoader)(
-      args: Seq[AnyRef]): AnyRef =
+  def newInstance(name: String, loader: ClassLoader)(args: Seq[AnyRef]): Any =
     newInstance(name, loader, Seq.fill(args.length)(null))(args)
 
-  def newInstance(name: String, loader: ClassLoader, paramTypes: Seq[Class[_]])(
-      args: Seq[Any]): AnyRef = {
+  def newInstance(fqcn: String, loader: ClassLoader, paramTypes: Seq[Class[_]])(
+      args: Seq[Any]): Any = {
```

Comment:
Same here: it should return `AnyRef`.

You'll need to cast the result of `.newInstance(args: _*)` to `AnyRef` for the code to typecheck. (such a cast always works, so it's fine)